Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java

```diff
@@ -2,13 +2,94 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.Context;
 import android.os.Bundle;
 
+import android.preference.PreferenceManager;
+
+import org.osmdroid.config.Configuration;
+import org.osmdroid.tileprovider.tilesource.TileSourceFactory;
+import org.osmdroid.views.MapView;
+
+
+
+import android.view.View;
+import android.widget.TextView;
+
+import com.teamapricot.projectwalking.handlers.CameraHandler;
+import com.teamapricot.projectwalking.photos.PhotoController;
+
+import com.teamapricot.projectwalking.Navigation.MapController;
+
 public class MainActivity extends AppCompatActivity {
+    private PhotoController photoController;
+    LocationHandler locationHandler;
+
+    MapView map = null;
+    private MapController mapController = null;
+
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
```

Comment:
This should be `protected` instead of `public`, since that is what the base class assumes. Otherwise it won't be overridden correctly. The base class in this case is simply the `Activity` class, see the link for details:

https://developer.android.com/reference/android/app/Activity.html#protected-methods

And here is the corresponding class inheritance:

https://developer.android.com/reference/androidx/appcompat/app/AppCompatActivity